src/BuilderPattern/Product/Vehicle.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace BuilderPattern.Product
+{
+    using System;
+    using Parts;
+
+    public class Vehicle
+    {
+        public Vehicle(VehicleType type, Brand brand, Engine engine, Wheel wheel, Seat seat)
+        {
+            this.VehicleType = type;
+            this.Brand = brand;
+            this.Engine = engine;
+            this.Seats = seat;
+            this.Wheels = wheel;
+        }
+
+        public VehicleType VehicleType { get; set; }
+
+        public Brand Brand { get; set; }
+
+        public Engine Engine { get; set; }
+
+        public Wheel Wheels { get; set; }
+
+        public Seat Seats { get; set; }
```

Comment:
Using the builder pattern allows us to have the logic of object creation in one single place so (which uses a single constructor). Therefore, I suggest marking these setters as private or even removing them altogether.

Reply:
I will make the setter private. At least someone can access the property to check its values.